reposerver/repository/repository.go

```diff
@@ -803,6 +807,9 @@ func (s *Service) newClientResolveRevision(repo *v1alpha1.Repository, revision s
 
 func (s *Service) newHelmClientResolveRevision(repo *v1alpha1.Repository, revision string, chart string) (helm.Client, string, error) {
```

Comment:
Please move this code into helm client and reuse in https://github.com/argoproj/argo-cd/blob/a6da0ca65b71472efedebe22afc338aeed9e268d/server/application/application.go#L1036

Reply:
out of scope of this PR

Reply:
this is tech debt, so I think out of scope